pkg/registry/registry.go

```diff
@@ -255,36 +254,43 @@ func (r *Registry) handleMirror(rw mux.ResponseWriter, req *http.Request, key st
 				return
 			}
 
-			// Modify response returns and error on non 200 status code and NOP error handler skips response writing.
-			// If proxy fails no response is written and it is tried again against a different mirror.
-			// If the response writer has been written to it means that the request was properly proxied.
-			succeeded := false
 			scheme := "http"
 			if req.TLS != nil {
 				scheme = "https"
 			}
-			u := &url.URL{
+			u := url.URL{
 				Scheme: scheme,
 				Host:   ipAddr.String(),
+				Path:   req.URL.Path,
+				// TODO: Should this error early if not set?
+				RawQuery: fmt.Sprintf("ns=%s", req.URL.Query().Get("ns")),
 			}
-			proxy := httputil.NewSingleHostReverseProxy(u)
-			proxy.Transport = r.transport
-			proxy.ErrorHandler = func(_ http.ResponseWriter, _ *http.Request, err error) {
-				log.Error(err, "proxy failed attempting next")
+			forwardReq, err := http.NewRequest(req.Method, u.String(), nil)
```

Comment:
Shouldn't the request body be forwarded? Or do we know that it's always empty?

If we know it's always empty, maybe use `http.NoBody` instead of nil and leave a comment about why.

Reply:
Right now this is a pull only registry so there will never be any request body. If data is sent it is probably wrong.

Reply:
Worth mentioning that the OCI distribution conformance tests are a part of the e2e tests, which gives some confidence in the changes being made.

Reply:
// An alternative, however, is to simply set [Request.Body] to nil.

In regards to the question of why not http.NoBody.